nixos/modules/config/qt.nix

```diff
@@ -0,0 +1,259 @@
+      }
+    //
+    optionalAttrs (cfg.style != null)
+      { widgetStyle = cfg.style.name; };
+  icons =
+    optionalAttrs (cfg.iconTheme != null)
+      { Theme = cfg.iconTheme.name; };
+
+  qt =
+    optionalAttrs (cfg.font != null)
+      { font = ''"${cfg.font.name}"''; }
+    //
+      { style = "GTK+"; };
+
+  fontType = types.submodule {
+    options = {
+      package = mkOption {
+        internal = true;
+        type = types.nullOr types.package;
+        default = null;
+      };
+      name = mkOption {
+        internal = true;
+        type = types.str;
+      };
+    };
+  };
+  themeType = types.submodule {
+    options = {
+      package = mkOption {
+        internal = true;
+        type = types.nullOr types.package;
+        default = null;
+      };
+      name = mkOption {
+        internal = true;
+        type = types.str;
+      };
+    };
+  };
+
+  optionalPackage = opt:
+    optional (opt != null && opt.package != null) opt.package;
+
+  platforms = {
+    gtk2 = rec {
+      description = ''
+        <varlistentry>
+          <term><literal>gtk2</literal></term>
+          <listitem><para>Use GTK2 theme with
+            <link xlink:href="https://github.com/qt/qtstyleplugins">qtstyleplugins</link>
+          </para></listitem>
+        </varlistentry>
+      '';
+      styles = [ "cleanlooks" "gtk2" "cde" "motif" "plastique" ];
+
+      assertions = [
+        {
+          assertion = cfg.style != null && any (name: name == cfg.style.name) styles;
+          message = "`qt5.style.name` is not one of [ ${toString styles} ].";
+        }
+        {
+          assertion = cfg.font == null && cfg.iconTheme == null;
+          message = "`qt.font` and `qt.iconTheme` are only supported by kde platform.";
+        }
+      ];
+      environment.variables.QT_QPA_PLATFORMTHEME = "gtk2";
+      environment.variables.QT_STYLE_OVERRIDE = cfg.style.name;
+      environment.systemPackages = [ pkgs.libsForQt5.qtstyleplugins ];
+    };
+    qgnomeplatform = {
+      description = ''
+        <varlistentry>
+          <term><literal>qgnomeplatform</literal></term>
+          <listitem><para>Use GNOME theme with
+            <link xlink:href="https://github.com/FedoraQt/QGnomePlatform">qgnomeplatform</link>
+          </para></listitem>
+        </varlistentry>
+      '';
+
+      assertions = [
+        {
+          assertion = cfg.font == null && cfg.iconTheme == null;
+          message = "`qt.font` and `qt.iconTheme` are only supported by kde platform.";
+        }
+      ];
+      environment.variables.QT_QPA_PLATFORMTHEME = "qgnomeplatform";
+      # TODO: make this optional once https://github.com/NixOS/nixpkgs/issues/54150 is fixed
+      # qgnomeplatform reads theme and other settings from dconf db
+      environment.variables.QT_STYLE_OVERRIDE = cfg.style.name;
+      environment.variables.XDG_DATA_DIRS = [ "${pkgs.gsettings-desktop-schemas}/share/gsettings-schemas/${pkgs.gsettings-desktop-schemas.name}" ];
```

Comment:
```suggestion
```

I'm going to merge https://github.com/NixOS/nixpkgs/pull/83245, so we won't need this.
It's also a bad idea to set XDG_DATA_DIRS in general like this.